### ISSUE TITLE
Off-policy Meta RL 알고리즘 작성

### ISSUE THREAD

**Issue by neverparadise** [Intelligent-Agent-Lab/ARC_PRIZE_2025#79]
FOR many iterations:
    Sample training tasks
    FOR each task:
        1. Adapt (k-shot)
        2. Evaluate
        3. Compute meta loss
    UPDATE θ using meta-gradient  ← 학습

```
class MetaRL:
    def __init__(self):
        self.θ = initialize_meta_parameters()
    
    def meta_train(self, train_tasks, num_iterations):
        """Meta-training loop"""
        for iter in range(num_iterations):
            task_batch = sample(train_tasks, batch_size=5)
            meta_loss = 0
            
            for task in task_batch:
                # Adaptation + Evaluation
                adapted_params = self.adapt(task, self.θ)
                task_loss = self.evaluate(task, adapted_params)
                meta_loss += task_loss
            
            # UPDATE meta-parameters
            self.θ = self.θ - β * grad(meta_loss, self.θ)  # ← 업데이트!
    
    def meta_test(self, test_tasks):
        """Meta-testing (evaluation only)"""
        results = []
        
        # θ는 고정! (self.θ 업데이트 없음)
        for task in test_tasks:
            # 동일한 adaptation + evaluation
            adapted_params = self.adapt(task, self.θ)
            performance = self.evaluate(task, adapted_params)
            results.append(performance)
        
        return mean(results)  # 성능 보고만
    
    def adapt(self, task, θ):
        """Inner-loop adaptation (train과 test에서 동일)"""
        context = collect_context(task, k_shot=5)
        z = self.context_encoder(context, θ)
        return z  # or adapted parameters
    
    def evaluate(self, task, adapted_params):
        """Evaluation (train과 test에서 동일)"""
        episodes = rollout_multiple(task, adapted_params, n=10)
        return mean([sum(ep.rewards) for ep in episodes])
```
